src/validators/ArrayValidator.ts

```diff
@@ -95,6 +110,26 @@ export class ArrayValidator<T> extends BaseValidator<T[]> {
 			? Result.ok(transformed)
 			: Result.err(new CombinedPropertyError(errors));
 	}
+
+	private handleUniqueStrategy(givenValues: unknown[]) {
+		if (givenValues.length < 2) return [];
+		const errors: [number, BaseError][] = [];
+		const values = [...givenValues];
+		let previousValue = values.shift();
+
+		for (let i = 0; i < values.length; i++) {
```

Comment:
Rather than doing this you can simply do:
```ts
const givenValues: unknown[] = [];

const possiblyUnique = new Set(givenValues);

if (possiblyUnique.size === givenValues.length) {
	// is unique
} else {
	// is not unique
}
```
Because `Set`s always eliminate all non-unique values when its size matches that of the input, the input was unique.

Reply:
If I use `Set`, I need to stringify it like [sapphiredev/shapeshift@`165b38b` (#141)](https://github.com/sapphiredev/shapeshift/pull/141/commits/165b38bc2d02f1afcae8e0990403f0f89dc03c04) this which isn't 100% accurate too. So I've reverted it to use `isDeepStrictEqual`. We can also use `fast-deep-equal` which is fater that `isDeepStrictEqual` lmk wdyt?

Reply:
Honestly the published version of fast-deep-equal is one of those typical very old poorly maintained packages that doesn't get releases "because it just works", for example see an issue like https://github.com/epoberezkin/fast-deep-equal/issues/43, and there are also issues such as https://github.com/epoberezkin/fast-deep-equal/issues/81.

Lastly they also ship React comparisons which is wholly unnecessary for us and would just add package bloat. 

Due to these reasons I think we should stick with `isDeepStrictEqual`.